src/components/app/check-font-loaded.js

```diff
@@ -0,0 +1,69 @@
+import { UPDATE_FONT_LOADED } from '../../actions';
+
+/**
+ * Prevent chart from displaying until the webfont has loaded,
+ * to ensure that text label SVGRect BBox measurements are accurate
+ * @param {Object} store Redux store
+ * @return {Promise} Resolves when font is detected as having loaded, or after 8 seconds
+ */
+const checkFontLoaded = store =>
+  new Promise(resolve => {
+    const { fonts } = document;
+    const fontName = '10px Titillium Web';
+    /**
+     * Dispatch an event to show the chart
+     */
+    const setLoaded = () => {
+      store.dispatch({ type: UPDATE_FONT_LOADED, fontLoaded: true });
+      resolve();
+    };
+
+    // If FontFaceSet is not supported then default to true immediately
+    if (!Boolean(fonts && fonts.check)) {
+      setLoaded();
+      return;
+    }
+
+    /**
+     * Check whether the webfont is ready, and show the chart if so
+     */
+    const checkIfLoaded = () => {
+      if (fonts.check(fontName)) {
+        // Add 0.1s delay because fonts.check often returns true too early
+        setTimeout(setLoaded, 100);
+      }
+    };
+
+    /**
+     * Recursive requestAnimationFrame step function, which acts as a backup
+     * to the native FontFaceSet event handlers.
+     */
+    const step = () => {
+      // If it's been 8 seconds since page load then just set loaded=true
```

Comment:
Hey, any specific reason why you set 8 seconds here?

Reply:
It's pretty arbitrary, but 8 seconds seemed like an appropriate length of time because it should be long enough that the font should have loaded by now even on slow-ish connections, but short enough that it won't cause users massive headaches.